### PR TITLE
Magento2

### PR DESCRIPTION
**Pull request by valguss**
PR to optimise the image cleaner so that it gets an array of all images in the DB rather than getting them one by one
Also a fix for the default values command

### CODE REVIEW

File: Console/Command/RemoveUnusedMediaCommand.php [hunk 1]
```diff
@@ -71,6 +71,16 @@ public function execute(InputInterface $input, OutputInterface $output) : void
 
         $directoryIterator = new \RecursiveDirectoryIterator($imageDir);
 
+        $imagesToKeepArray = $connection->fetchAll(
```
Comment (sprankhub) [inline Console/Command/RemoveUnusedMediaCommand.php]: I think you could save the `foreach` by using `fetchCol` here.

File: Console/Command/RestoreUseDefaultValueCommand.php [hunk 2]
```diff
@@ -52,6 +67,8 @@ public function execute(InputInterface $input, OutputInterface $output)
         $counts = array();
         $i = 0;
         $tables = array('varchar', 'int', 'decimal', 'text', 'datetime');
+        $column = $this->productMetaData->getEdition() === 'Enterprise' ? 'row_id' : 'entity_id';
```
Comment (sprankhub) [inline Console/Command/RestoreUseDefaultValueCommand.php:70]: I do not have access to the Enterprise edition. Could anyone confirm this?

**Review by sprankhub — COMMENTED** [review]
Thanks a lot for your PR, @valguss! I left two small comments :)

File: Console/Command/RestoreUseDefaultValueCommand.php (hunk 2 above)
Comment (valguss) [reply Console/Command/RestoreUseDefaultValueCommand.php:70]: This is where I got the original code from: https://magento.stackexchange.com/questions/134265/magento-2-programmatically-check-magento-version-ce-ee

File: Console/Command/RemoveUnusedMediaCommand.php (hunk 1 above)
Comment (valguss) [reply Console/Command/RemoveUnusedMediaCommand.php]: Code now updated. Thanks for the hint. Hadn't actually used fetchCol before

File: Console/Command/RestoreUseDefaultValueCommand.php (hunk 2 above)
Comment (sprankhub) [reply Console/Command/RestoreUseDefaultValueCommand.php:70]: I meant if it is really called `row_id` in the Enterprise edition. But I just believe you :)

**Pull request merged**